cmd/root.go

```diff
@@ -18,22 +18,65 @@ import (
 	"fmt"
 	"os"
 
+	"github.com/gophergala2016/reviewer/reviewer"
 	"github.com/spf13/cobra"
 	"github.com/spf13/viper"
 )
 
 var cfgFile string
 
-// This represents the base command when called without any subcommands
+type config struct {
+	Authorization struct {
+		Token string
+	}
+	Repositories map[string]struct {
+		Username string
+		Status   bool
+		Required int
+	}
+}
+
+// RootCmd represents the base command when called without any subcommands
 var RootCmd = &cobra.Command{
 	Use:   "reviewer",
 	Short: "Code review your pull requests",
 	Long: `By running reviewer your repo's pull requests will get merged
 according to the configuration file.`,
-    Run: func(cmd *cobra.Command, args []string) {
-        // TODO: Work your own magic here
-        fmt.Println("root called")
-    },
+	Run: func(cmd *cobra.Command, args []string) {
+		var C config
+		err := viper.Unmarshal(&C)
+		if err != nil {
+			fmt.Printf("Error parsing configuration %v", err)
+			return
+		}
+		client, err := reviewer.GetClient()
+		if err != nil {
+			fmt.Printf("Error creating GitHub client %v", err)
+			return
+		}
+
+		//TODO: validate imput parameters (e.g. Requiered = 0)
+		for repoName, repoParams := range C.Repositories {
+			if repoParams.Status == false {
+				fmt.Printf("- %v/%v Discarded (repo disabled)\n", repoParams.Username, repoName)
+				continue
+			}
+			prInfos, err := reviewer.GetPullRequestInfos(client, repoParams.Username, repoName)
+			if err != nil {
+				fmt.Printf("Error getting pull request info of repo %v/%v", repoParams.Username, repoName)
+				return
+			}
+			fmt.Printf("+ %v/%v\n", repoParams.Username, repoName)
+			for _, prInfo := range *prInfos {
+				if prInfo.Score >= repoParams.Required {
+					fmt.Printf("  + %v MERGE (%v) score %v of %v required\n", prInfo.Number, prInfo.Title, prInfo.Score, repoParams.Required)
+					// merge here
+				} else {
+					fmt.Printf("  - %v NOP   (%v) score %v of %v required\n", prInfo.Number, prInfo.Title, prInfo.Score, repoParams.Required)
+				}
+			}
+		}
+	},
```

Comment:
I think, this code should be implemented in a function in `reviewer` package, so it could be tested appropriately.